vision/benchmarks/accuracy/bench_text.py

```diff
@@ -2,32 +2,152 @@
+
 from text.detect_words import TextDetector
 
+IMG_FOLDER = "text"
 
-IMG_FOLDER = 'boat'
 
 class AccuracyRussianWord:
     """
-    Testing the text detector.
+    Measuring accuracy of the text detector.
     """
-    def setup(self):
-        pass
 
-    def accuracy_detector(self, color_image, depth_image):
+    def __init__(self):
+        self.setup()
+
+    def setup(self) -> None:
+        """
+        Benchmark setup.
+
+        Returns
+        -------
+        None
+        """
+        self.text_detector = TextDetector()
+
+    def accuracy_detector(
+        self, color_image: np.ndarray, depth_image: np.ndarray
+    ) -> list:
         """
         Accuracy of detectRussianWord.
 
         Returns
         -------
         List[BoundingBox]
         """
-        bounding_boxes = TextDetector().detect_russian_word(color_image, depth_image)
+        return self.text_detector.detect_russian_word(
+            color_image=color_image, depth_image=depth_image
+        )
+
+
+class BenchTextAccuracy:
+    """
+    Object for storing parameters of and running the text accuracy benchmark.
+
+    Parameters
+    ----------
+    plot_text: bool
+        Whether to plot the text on the image and save.
+    """
+
+    def __init__(self, plot_text: bool = False):
+        self.OUTPUT_IMGS_DIR = (
+            "marked_images"  # Folder to output saved images to if necessary
+        )
+        self.OUTPUT_RESULTS_DIR = (
+            "results"  # Folder to output resulting BoundingBoxes to
+        )
+
+        self.plot_text = plot_text
+
+        if not os.path.isdir(self.OUTPUT_IMGS_DIR):
+            os.mkdir(self.OUTPUT_IMGS_DIR)
+        if not os.path.isdir(self.OUTPUT_RESULTS_DIR):
+            os.mkdir(self.OUTPUT_RESULTS_DIR)
+
+    def set_parameters(self, plot_text: bool = False) -> None:
+        """
+        Sets the parameters for running the benchmark.
+
+        Parameters
+        ----------
+        plot_text: bool
+            Whether to plot the text on the image and save.
+
+        Returns
+        -------
+        None
+        """
+        self.plot_text = plot_text
+        return
+
+    def bench_accuracy(
+        self,
+        file_output: IOBase,
+        tester: AccuracyRussianWord,
+        image: np.ndarray,
+        depth: np.ndarray,
+        filename: str,
+    ) -> bool:
+        """
+        Runs all text detection accuracy benchmarks on an image. Outputs results to csv file.
+
+        Parameters
+        ----------
+        file_output: IO
+            File stream to output results to.
+        tester: AccuracyRussianWord
+            Benchmark class to use. Initialized elsewhere to be static between image runs.
+        image: np.ndarray
+            The color image from the frame.
+        depth: np.ndarray
+            The depth image from the frame.
+        filename: str
+            The name of the image file.
+
+        Returns
+        -------
+        bool - whether all tests were completed.
```

Comment:
This could be more specific: from what I can tell it returns `True` if it doesn't complete and returns `False` if it doesn't crash at any point.